### PR TITLE
ruby upgrade 3.2.2

### PR DESCRIPTION
**Pull request by bo-aaf**
Upgrade ruby version. I have only gotten this to pass with unit test. Let me know if there is more to do with testing this ...

### CODE REVIEW

File: app/models/concerns/raw_entity_descriptor_deconstructor.rb [hunk 1]
```diff
@@ -138,9 +138,11 @@ def extract_single_sign_on_services(single_sign_on_services_node)
 
   private
 
+  # rubocop:disable Style/OpenStructUse
   def os(hash)
```
Comment (bo-aaf) [inline app/models/concerns/raw_entity_descriptor_deconstructor.rb:142]: This data structure is very tightly coupled with the test. This should be in a separate pr

File: app/models/entity_descriptor.rb [hunk 2]
```diff
@@ -80,23 +80,25 @@ def entity_attribute?
   def technical_contact_count
     ContactPerson.join(:entity_descriptors, id: :entity_descriptor_id)
                  .where(Sequel.qualify(:entity_descriptors, :id) => id)
-                 .and(Sequel.qualify(:contact_people, :contact_type_id) =>
+                 .where(Sequel.qualify(:contact_people, :contact_type_id) =>
```
Comment (phyzical) [inline app/models/entity_descriptor.rb:83]: maybe this?

File: app/models/localized_uri.rb [hunk 3]
```diff
@@ -1,7 +1,7 @@
 # frozen_string_literal: true
 
 class LocalizedURI < Sequel::Model
-  plugin :class_table_inheritance
+  plugin :class_table_inheritance, ignore_subclass_columns: %i[created_at updated_at]
```
Comment (phyzical) [inline app/models/localized_uri.rb:4]: maybe these changes?

Comment (bo-aaf) [reply app/models/localized_uri.rb:4]: Yea the sequel 5 is throwing this error
```
Sequel::Error: class_table_inheritance with duplicate column names (other than the primary key column) is not supported, make sure tables have unique column names (duplicate columns: [:created_at :updated_at]). If this is desired, specify these columns in the :ignore_subclass_columns option when initializing the plugin
```

Kinda in a conundrum now, Ruby 3 couldnt work with sequel 4, so i had to use 5. Lemme see if i can figure something out..

Comment (bo-aaf) [reply app/models/localized_uri.rb:4]: And yes sequel 5 gem is causing this to error somehow in docker

**Review by matthew-puku — APPROVED** [review]

**Comment by matthew-puku** [comment]
I think try testing this is dev with /deploy.

**Review by waldofouche — APPROVED** [review]
+1 test in dev

**Comment by bo-aaf** [comment]
tested manually in dev! this works

File: Gemfile [hunk 4]
```diff
@@ -26,7 +26,7 @@ gem 'jbuilder'
 
 gem 'bcrypt', '~> 3.1'
 
-gem 'rugged', '0.28.4.1', require: false
+gem 'rugged', '~> 1.1', require: false
```
Comment (vladimir-mencl-eresearch) [inline Gemfile:29]: Just FYI @phyzical , this broke our sync-to-git deployment - because the Rugged library stopped recognising git+ssh remote repositories: `unsupported URL protocol (Rugged::NetworkError)`

I had to change our deployment to:
* run `bundle config build.rugged --with-ssh` before installing the Gems (to force Rugged to be installed with SSH support).
* preinstall also the `ssh-rsa` host key for the remote host (besides the `ecdsa-sha2-nistp256` key we'd otherwise rely on) because the libssh2/libgit2 libraries used by Rugged only use RSA ssh host keys.

Already fixed it for our deployment - posting here so that you are aware of it in case you wanted to use this functionality ... and don't have to reinvent the wheel.

Cheers,
Vlad
